compiler/modules/front/codeGen.bal

```diff
@@ -624,6 +651,86 @@ function codeGenAssignToMember(CodeGenContext cx, bir:BasicBlock startBlock, Env
     return { block: nextBlock };
 }
 
+function codeGenCompoundAssignToMember(CodeGenContext cx, bir:BasicBlock bb, Environment env, MemberAccessLExpr lValue, Expr rexpr, CompoundAssignOp op, err:Position pos) returns CodeGenError|StmtEffect{
+    //TODO Move to functions
```

Comment:
This is far too much code.

Reply:
I can move the different instructions to functions, and use those functions within codeGenExpr too. Is that enough or do I need to change how instructions are created here?

Reply:
As I think you're saying, you need to factor out the code from codeGenExpr (at the point where it deals with bir:Operand).

I don't know whether that will be enough. But when you're doing it right, there won't be a lot of code.

Reply:
I have just realized that it makes no sense to implement compound assign to member in this subset, because we only have arrays and maps of type `any` and there are no binary operators that work with `any`.

So for this pull request, just implement compound assignment where the LHS is a variable reference expression.